tap_harvest/client.py

```diff
@@ -0,0 +1,146 @@
+            return float(config_request_timeout)
+        raise Exception(
+            "The entered timeout is invalid, it should be a valid none-zero integer.")
+
+    # backoff for Timeout error is already included in "requests.exceptions.RequestException"
+    # as it is a parent class of "Timeout" error
+    @backoff.on_exception(backoff.expo,
+                          requests.exceptions.RequestException,
+                          max_tries=5,
+                          giveup=lambda e: e.response is not None and 400 <= e.response.status_code < 500,
+                          factor=2)
+    def _refresh_access_token(self):
+        """
+        Create an access token using the refresh token.
+        """
+        LOGGER.info("Refreshing access token")
+        resp = self.session.request('POST',
+                                    url=BASE_ID_URL + 'oauth2/token',
+                                    data={
+                                        'client_id': self._client_id,
+                                        'client_secret': self._client_secret,
+                                        'refresh_token': self._refresh_token,
+                                        'grant_type': 'refresh_token',
+                                    },
+                                    headers={"User-Agent": self._user_agent})
+        expires_in_seconds = resp.json().get('expires_in', 17 * 60 * 60)
+        self._expires_at = pendulum.now().add(seconds=expires_in_seconds)
+        resp_json = {}
+        try:
+            resp_json = resp.json()
+            self._access_token = resp_json['access_token']
+        except KeyError as key_err:
+            if resp_json.get('error'):
+                LOGGER.critical(resp_json.get('error'))
+            if resp_json.get('error_description'):
+                LOGGER.critical(resp_json.get('error_description'))
+            raise key_err
+        LOGGER.info("Got refreshed access token")
+
+    def get_access_token(self):
+        """
+        Return access token if available or generate one.
+        """
+        if self._access_token is not None and self._expires_at > pendulum.now():
+            return self._access_token
+
+        self._refresh_access_token()
+        return self._access_token
+
+    def get_account_id(self):
+        """
+        Get the account Id of the Active Harvest account.
+        It will throw an exception if no active harvest account is found.
+        """
+        if self._account_id is not None:
+            return self._account_id
+
+        response = self.session.request('GET',
+                                        url=BASE_ID_URL + 'accounts',
+                                        headers={'Authorization': 'Bearer ' + self._access_token,
+                                                 'User-Agent': self._user_agent},
+                                        timeout=self.request_timeout)
+
+        if response.json().get('accounts'):
```

Comment:
Add comment

Reply:
Added comment to explain `if` condition.